hpy/devel/include/hpy.h

```diff
@@ -37,6 +72,28 @@ extern "C" {
 #  define _HPy_NO_RETURN
 #endif
 
+
+// clang and gcc supports __has_attribute, MSVC doesn't. This should be enough
```

Comment:
What if we surround the auto-generated trampolines with some `#ifdef HPY_ABI_UNIVERSAL` ?

Reply:
Not to this particular line but to the block from line 78 to 82.
What I meant is: `_HPY_LEGACY` is used to mark legacy functions such that the extension author gets compilation error if they are called and compiled in universal ABI, right?
My suggestion was that instead of this attribute, we could just define those functions in a way that the produce a compilation error. But I'm fine with this approach.

Reply:
ah ok I understand what I mean. I think it would be worse, because in that case the C compiler would assume that e.g. `HPy_AsPyobject` has an implicit declaration: it would emit a warning and happily compile, which is even worse.